dynamics/rolldowndata/rolldown/clean_rolldown_data.py

```diff
@@ -0,0 +1,90 @@
+# files to be cleaned should be put into
+# /dynamics/rolldowndata/rolldown/unprocessed
+import pandas as pd
+from os import listdir, rename
+from os.path import isfile, join
+
+# set folder paths
+unprocessed_path = 'unprocessed'
+processed_path = 'processed'
+raw_path = 'raw'
+clean_path = 'clean'
+
+# make list of unprocessed files
+only_files = [f for f in listdir(unprocessed_path) if isfile(join(unprocessed_path, f))]
+
+# create file paths
+file_paths = []
+for f in only_files:
+    f = join('.', unprocessed_path, f)
+    file_paths.append(f)
+
+# clean and transform each file in file_paths list
+for i, f in enumerate(file_paths):
+    rolldown = pd.read_csv(f)
+
+    # make a new boolean column, needed to clean data later
+    rolldown['throttle and brake == 0 or na'] = ((rolldown['throttle'] == 0) & \
+                                            (rolldown['mechanical_brake_state'] == 0)) | \
+                                            ((rolldown['throttle'].isna()) & \
+                                             (rolldown['mechanical_brake_state'].isna()))
+
+    # resort index to ensure data is in correct order
+    # then reset the index, finally, make the index a column
+    # making the index a column makes it easier to find the subset we need
+    rolldown.reset_index(drop = True, inplace = True)
+    rolldown.reset_index(inplace = True)
+    rolldown.rename(columns={'index' : 'row_no'}, inplace = True)
+
+    # remove unnecessary columns
+    # include throttle and brake state for now
+    col_snip = ['row_no', 'throttle', 'mechanical_brake_state', 'Time Offset [ms]', 'vehicle_velocity_left', \
+                    'vehicle_velocity_right', 'throttle and brake == 0 or na']
+    rolldown = rolldown[col_snip]
+
+    # make new dataframe to find subset of rows
+    # that contain the required data
+    find_subset = pd.concat([rolldown['throttle and brake == 0 or na'], rolldown['row_no']], axis = 1)
+
+    # cumsum on boolean column to find multiple
+    # subsets of rows fitting our requirements
+    find_subset_bool = find_subset['throttle and brake == 0 or na'] != find_subset['throttle and brake == 0 or na'].shift()
+    find_subset_cumsum = find_subset_bool.cumsum()
+    subset_groups = find_subset.groupby(find_subset_cumsum)
+    group_counts = subset_groups.agg({'row_no':['count', 'min', 'max']})
+
+    # drop unnecessary column levels
+    group_counts.columns = group_counts.columns.droplevel()
```

Comment:
what columns does this drop?

Reply:
before this line it had two rows of column headers, I drop one level